addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -292,7 +292,11 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		// Sometimes it is unable to retrieve the complete list of server APIs i.g. an aggregated API server is down,
+		// ErrGroupDiscoveryFailed is returned in this case. We can still get other API resources and ignore this error.
```

Comment:
Remove unecessary comments. They might make sense to you right now but someone coming in 6 months later they are out of context, and is something else to mantain. 
Add any explanation of why we are ignoring the error to  logging. (info, debut, or, etc...)

Reply:
point taken.